test/cppsuite/test_harness/workload_generator.h

```diff
@@ -96,67 +95,135 @@ class workload_generator {
                 testutil_check(cursor->insert(cursor));
             }
         }
-        debug_info(std::to_string(collection_count) + " key/value inserted", DEBUG_INFO);
+        debug_info(
+          std::to_string(collection_count) + " key/value inserted", _trace_level, DEBUG_INFO);
+        debug_info("Load stage done", _trace_level, DEBUG_INFO);
         return (0);
     }
 
     /* Do the work of the main part of the workload. */
     int
     run()
     {
-        /* Empty until thread management lib is implemented. */
-        return (0);
+
+        WT_SESSION *session;
+        int64_t duration_seconds, read_threads;
+
+        session = nullptr;
+        duration_seconds = read_threads = 0;
+
+        testutil_check(_configuration->get_int(DURATION_SECONDS, duration_seconds));
+        testutil_check(_configuration->get_int(READ_THREADS, read_threads));
+
+        /* Create a separate read thread for each collection. It's likely that this will change. */
```

Comment:
This comment is wrong, I know I did it originally but can you update it to be accurate?